server/src/main/java/org/candlepin/resource/ActivationKeyResource.java

```diff
@@ -279,9 +302,28 @@ public ActivationKey removeProductIdFromKey(
      */
     @GET
     @Produces(MediaType.APPLICATION_JSON)
-    public List<ActivationKey> findActivationKey() {
-        List<ActivationKey> keyList = activationKeyCurator.listAll();
-        return keyList;
+    public Response findActivationKey() {
+        // TODO: Replace this with use of a cursor/iterator
```

Comment:
Not really. There will probably be changes to be made there, and the TODO called out here would likely be touched up as part of a bigger patch that addresses several similar calls -- especially in light of some performance details we got today.